syntax = "proto3";

package vectra.telemetry;

message GpsTrace {
  string driver_id = 1;
  string vehicle_id = 2;
  double latitude = 3;
  double longitude = 4;
  double altitude = 5;
  double speed_mps = 6;     // Meters per second
  double bearing = 7;       // 0-360 degrees
  double accuracy_m = 8;    // GPS accuracy in meters
  int64 timestamp_ms = 9;   // Unix epoch millis
  string event_type = 10;   // "PING", "STOP", "SCAN", "NAV_START"
  string battery_level = 11;
}